packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -317,6 +317,21 @@ options:
         }
       }
     }
+    for (Api api in root.apis) {
+      for (Method method in api.methods) {
+        if (_validTypes.contains(method.argType)) {
+          result.add(Error(
+              message:
+                  'Unsupported argument type: "${method.argType}" in API: "${api.name}" method: "${method.name}'));
```

Comment:
This message might be a bit confusing when users get a bool isn't supported error. Maybe add some context like "primitives aren't yet supported, see feature request: https://github.com/..."

Reply:
Ugh yea.  I don't want to commit to supporting it though either.

Reply:
ah ok, SG